app/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepository.scala

```diff
@@ -79,7 +80,7 @@ class CaseMongoRepository @Inject()(mongoDbProvider: MongoDbProvider, jsonMapper
   }
 
   override def update(c: Case): Future[Option[Case]] = {
-    updateDocument(selector = jsonMapper.fromReference(c.reference), update = c)
+    updateDocument(selector = jsonMapper.fromReference(c.reference), update = c, upsert = appConfig.upsertPermitted)
```

Comment:
if you introduce the `upsert` flag here, the `insert` method above is useless.
you can have a single `upsert` method doing both `insert` and `update`

Reply:
Same comment applies to the service layer